build.gradle.kts

```diff
@@ -28,14 +28,14 @@ configurations.all {
 
 dependencies {
     implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlinVersion")
-    api("org.testcontainers:testcontainers:1.15.1")
+    api("org.testcontainers:testcontainers:1.17.3")
```

Comment:
Is `testcontainers:1.17.3` incompatible with `testcontainers:1.15.1` and that's why we need to change `SshUbuntuContainer`?

Reply:
Ah, no, the rationale is in the commit message :+1: 

> To resolve conflicting dependency net.java.dev.jna:jna: 5.2.0 vs 5.5.0
in ssh library which is consumer of this library

So even if jna:5.5.0 is compatible with jna:5.2.0, then we're getting bogus conflicts. But which lib is declaring compatibility with jna:5.2.0, while not declaring it with jna:5.5.0?

Reply:
It seems com.github.docker-java:docker-java-transport-zerodep in ssh-ubuntu.
ssh-ubuntu hasn't been updated in ssh library for a long time, there is 0.1.0 version

```
❯ cd ssh-ubuntu
❯ ./gradlew -q dependencies
+--- org.testcontainers:testcontainers:1.15.1
|    +--- junit:junit:4.12
|    |    \--- org.hamcrest:hamcrest-core:1.3
|    +--- org.slf4j:slf4j-api:1.7.30 -> 1.7.25
|    +--- org.apache.commons:commons-compress:1.20
|    +--- org.rnorth.duct-tape:duct-tape:1.0.8
|    +--- org.rnorth.visible-assertions:visible-assertions:2.1.2
|    |    \--- net.java.dev.jna:jna:5.2.0 -> 5.5.0
|    +--- com.github.docker-java:docker-java-api:3.2.7
|    |    +--- com.fasterxml.jackson.core:jackson-annotations:2.10.3
|    |    \--- org.slf4j:slf4j-api:1.7.30 -> 1.7.25
|    \--- com.github.docker-java:docker-java-transport-zerodep:3.2.7
|         +--- com.github.docker-java:docker-java-transport:3.2.7
|         +--- org.slf4j:slf4j-api:1.7.25
|         \--- net.java.dev.jna:jna:5.5.0
```

Reply:
```
|    +--- org.rnorth.visible-assertions:visible-assertions:2.1.2
|    |    \--- net.java.dev.jna:jna:5.2.0 -> 5.5.0
```
They are probably compatible with 5.5.0, but too lazy to express that via dependency range declarations.

So this PR is trying to find a matching permutation of versions, which do not suffer from such a bogus conflict. It's a lot of work on our (consumer) side, while the lib producer is the cause. We already paid the development price here, so we can merge, but for the future we can use other, cheaper or maintainable solutions:
* remove dependency on a lazy lib
* report the overly strict dependency to the lib authors
* contribute a dependency relaxation fix to the lib
* manually ignore/resolve the bogus conflict via `useVersion` in `build.gradle.kts` (we chose this one most often)